ravenframework/CustomModes/SlurmSimulationMode.py

```diff
@@ -190,27 +191,43 @@ def remoteRunCommand(self, runInfoDict):
     assert self.__runSbatch and not self.__inSlurm
     return self.__createAndRunSbatch(runInfoDict)
 
-  def XMLread(self, xmlNode):
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = InputData.parameterInputFactory("mode", ordered=False, contentType=InputTypes.StringType)
+    inputSpecification.addSub(InputData.parameterInputFactory("runSbatch"))
+    inputSpecification.addSub(InputData.parameterInputFactory("memory", contentType=InputTypes.StringType))
+    inputSpecification.addSub(InputData.parameterInputFactory("coresneeded", contentType=InputTypes.IntegerType))
+    inputSpecification.addSub(InputData.parameterInputFactory("partition", contentType=InputTypes.StringType))
+    inputSpecification.addSub(InputData.parameterInputFactory("MPIParam", contentType=InputTypes.StringType))
+    inputSpecification.addSub(InputData.parameterInputFactory("noprecommand"))
```

Comment:
any specific reason for naming here? some are low cases, some using upper cases for the Abbreviation.

Reply:
Hm, mostly from how they were in the manual (though the manual has noPrecommand, so that should be made consistent.).